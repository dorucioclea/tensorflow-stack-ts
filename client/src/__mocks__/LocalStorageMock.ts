import "jest";

class LocalStorageMock {
  store = {};
  setItem = jest.fn((key: string, val: {}) => Object.assign(this.store, { [key]: val }));
  getItem = jest.fn((key: string) => this.store[key]);
  clear = jest.fn(() => this.store = {});
}

Object.assign(window, { localStorage: new LocalStorageMock() });
module.exports = window.localStorage;
