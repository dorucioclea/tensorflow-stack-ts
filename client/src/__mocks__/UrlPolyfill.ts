// polyfill needed because we're using older JEST that is missing some
// URL functionality like url.searchParams

if (new URL("http://a/b").searchParams === undefined) {
    // tslint:disable-next-line:no-var-requires
    require("url-polyfill-light");
}
