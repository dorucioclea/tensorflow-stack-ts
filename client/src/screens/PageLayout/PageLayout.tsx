import * as React from "react";
import { renderRoutes } from "react-router-config";
import { Layout } from "antd";
import { Sidebar } from "../Sidear/Sidebar";
import { Header } from "../Header/Header";
import { sidebarRoutes } from "../../routes";
import "./PageLayout.less";

export const PageLayout = () => {
    return (
        <Layout className="ant-layout-has-sider">
            <Sidebar />
            <Layout>
                <Layout.Content>
                    <Header />
                    {/* <Redirect to="/" /> */}
                    {renderRoutes(sidebarRoutes)}
                </Layout.Content>
            </Layout>
        </Layout>
    );
};
