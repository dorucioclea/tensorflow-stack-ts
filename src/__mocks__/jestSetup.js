// to have a cleaner console output during test run we only really want to
// have the jest expect functions show as errors.
console.error = console.warn;
