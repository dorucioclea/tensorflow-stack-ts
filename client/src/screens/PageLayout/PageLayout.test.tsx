// Jest snapshot test
import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { StaticRouter } from "react-router";
import { PageLayout } from "./PageLayout";
import "../../__mocks__/LocalStorageMock";

describe("PageLayout testing", () => {

    test("Layout match to snapshot", () => {
        const component = TestRenderer.create(
            <StaticRouter location="someLocation" context={{}}>
                <PageLayout />
            </StaticRouter>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
