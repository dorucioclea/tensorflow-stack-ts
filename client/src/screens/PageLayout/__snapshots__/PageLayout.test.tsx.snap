// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout testing Layout match to snapshot 1`] = `
<div
  className="ant-layout-has-sider ant-layout ant-layout-has-sider"
>
  <div
    className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-has-trigger"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      />
      <ul
        className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
        role="menu"
        style={Object {}}
      >
        <li
          aria-disabled={undefined}
          className="ant-menu-item ant-menu-item-selected"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <i
              className="anticon anticon-dashboard"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="dashboard"
                fill="currentColor"
                height="1em"
                onClick={undefined}
                style={undefined}
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 385.6a446.7 446.7 0 0 0-96-142.4 446.7 446.7 0 0 0-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 0 0-142.4 96 446.7 446.7 0 0 0-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 0 1 140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 0 0-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 0 0 0 79.2 55.95 55.95 0 0 0 79.2 0 55.87 55.87 0 0 0 14.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 0 0-11.3 0l-56.6 56.6a8.03 8.03 0 0 0 0 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 0 0-11.3 0l-31.1 31.1a8.03 8.03 0 0 0 0 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </i>
            <span
              className="nav-text"
            >
              Dashboard
            </span>
          </a>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <a
            href="/todo"
            onClick={[Function]}
          >
            <i
              className="anticon anticon-check-square-o"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="check-square"
                fill="currentColor"
                height="1em"
                onClick={undefined}
                style={undefined}
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M433.1 657.7a31.8 31.8 0 0 0 51.7 0l210.6-292c3.8-5.3 0-12.7-6.5-12.7H642c-10.2 0-19.9 4.9-25.9 13.3L459 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H315c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8z"
                />
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                />
              </svg>
            </i>
            <span
              className="nav-text"
            >
              Todo
            </span>
          </a>
        </li>
        <li
          aria-disabled={undefined}
          className="ant-menu-item"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={undefined}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={undefined}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          title={undefined}
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            <i
              className="anticon anticon-file"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="file"
                fill="currentColor"
                height="1em"
                onClick={undefined}
                style={undefined}
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0 0 42 42h216v494z"
                />
              </svg>
            </i>
            <span
              className="nav-text"
            >
              About
            </span>
          </a>
        </li>
      </ul>
      <div
        className="sider-trigger"
      >
        <i
          className="anticon anticon-menu-fold trigger"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="menu-fold"
            fill="currentColor"
            height="1em"
            onClick={undefined}
            style={undefined}
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
            />
          </svg>
        </i>
      </div>
    </div>
    <div
      className="ant-layout-sider-trigger"
      onClick={[Function]}
      style={
        Object {
          "width": "200px",
        }
      }
    >
      <i
        className="anticon anticon-left"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="left"
          fill="currentColor"
          height="1em"
          onClick={undefined}
          style={undefined}
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    className="ant-layout"
  >
    <div
      className="ant-layout-content"
    >
      <div
        className="ant-layout-header"
      >
        <div
          className="ant-row-flex ant-row-flex-end ant-row-flex-middle"
          style={undefined}
        >
          <div
            className="ant-col-3"
            style={undefined}
          >
            <button
              className="ant-btn user-login ant-btn-primary"
              onClick={[Function]}
              title={undefined}
              type="button"
            >
              <i
                className="anticon anticon-user"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="user"
                  fill="currentColor"
                  height="1em"
                  onClick={undefined}
                  style={undefined}
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </i>
              <span>
                Log In
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
